packages/server/src/arpa_reporter/routes/audit-report.js

```diff
@@ -4,9 +4,42 @@ const express = require('express');
 
 const router = express.Router();
 
-const { requireUser } = require('../../lib/access-helpers');
+const { requireUser, getAdminAuthInfo } = require('../../lib/access-helpers');
 const audit_report = require('../lib/audit-report');
 const { useUser } = require('../use-request');
+const aws = require('../lib/aws-client');
+
+router.get('/:periodId/:filename', async (req, res) => {
+    let user;
+    try {
+        const info = await getAdminAuthInfo(req);
+        user = info.user;
+    } catch (error) {
+        res.redirect(encodeURI(`${process.env.WEBSITE_DOMAIN}/arpa_reporter/login?redirect_to=/api/audit_report/${req.params.periodId}/${req.params.filename}&message=Please login to visit the link.`));
+        return;
+    }
+
+    const s3 = aws.getS3Client();
+    const Key = `${user.tenant_id}/${req.params.periodId}/${req.params.filename}`;
+    const baseParams = { Bucket: process.env.AUDIT_REPORT_BUCKET, Key };
+
+    try {
+        await s3.headObject(baseParams).promise();
+    } catch (error) {
+        console.log(error);
+        res.redirect(`${process.env.WEBSITE_DOMAIN}/arpa_reporter?alert_text=The audit report you requested does not exist. Please try again by clicking the 'Send Audit Report By Email'.&alert_level=err`);
```

Comment:
Not sure if this is URL-encoded by default; if not, it may need to be.

Reply:
Additionally, it may be worth replacing "does not exist" with "has expired". It's a small change, but I think it communicates the situation a bit more directly and possibly has a lower chance of being misinterpreted.

Reply:
Done 👍